src/libraries/System.Reflection.Emit/tests/AssemblyBuilderTests.cs

```diff
@@ -424,7 +423,16 @@ void Invoke_Private_SameAssembly_ThrowsMethodAccessException()
 	    Assert.Throws<MethodAccessException>(() => d ());
 	}
 
+    [Fact]
+    public void DefineDynamicAssembly_InternalAssemblyLocationIsEmpty()
+    {
+        AssemblyBuilder assembly = Helpers.DynamicAssembly(nameof(DefineDynamicAssembly_InternalAssemblyLocationIsEmpty));
+        Assembly internalAssemblyBuilder  = AppDomain.CurrentDomain.GetAssemblies()
```

Comment:
Please also add:
```C#
Assert.IsEmpty(assembly.Location);
```

as that will test the change is `AssemblyBuilder` itself. (the check below tests the `InternalAssemblyBuilder` on CoreCLR, on Mono the two checks are the same, but that's OK).